src/main/java/org/fxmisc/wellbehaved/event/experimental/EventPattern.java

```diff
@@ -59,24 +59,27 @@
         return onlyIf(condition.negate());
     }
 
-    static <T extends Event, U extends T, V extends U, W extends U> EventPattern<T, U> anyOf(
-            EventPattern<T, V> p1, EventPattern<T, W> p2) {
+    @SafeVarargs
+    static <T extends Event, U extends T> EventPattern<T, U> anyOf(EventPattern<T, U>... events) {
```

Comment:
Can it be changed to

```
anyOf(EventPattern<T, ? extends U>... );
```

i.e. add `? extends` to the `U` parameter (only for the argument, not the return type).